vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java

```diff
@@ -50,14 +50,18 @@ public <T, E extends Exception> T wait(FailableSupplier<T, E> valueProvider, Pre
             long iterationStartTime = System.currentTimeMillis();
             value = valueProvider.get();
             long iterationEndTime = System.currentTimeMillis();
+            if (stopCondition.test(value))
+            {
+                break;
+            }
```

Comment:
```suggestion
            if (stopCondition.test(value))
            {
                break;
            }
            long iterationEndTime = System.currentTimeMillis();
```

Reply:
This is how it was at the beginning
And we again get **this** error

![image](https://github.com/vividus-framework/vividus/assets/58571912/f1a065df-e64f-47a3-9651-aad605d6686f)

Reply:
I guess this test is needed to be tuned

Reply:
this is expected, since we're changing the underneath logic